agent/grpc-middleware/auth_interceptor.go

```diff
@@ -75,6 +79,7 @@ func restrictPeeringEndpoints(authInfo credentials.AuthInfo, peeringSNI string,
 	if !ok {
 		return status.Error(codes.Unauthenticated, "invalid transport credentials")
 	}
+
 	if tlsAuth.State.ServerName == peeringSNI {
```

Comment:
I think we should probably move the check to down here, but otherwise this seems okay.

```
if peeringSNI != "" && peeringSNI == tlsAuth.State.ServerName
```

Reply:
Or just move it as-is further down.

```
if peeringSNI == "" {
  return nil
}
if peeringSNI == tlsAuth.State.ServerName {
```